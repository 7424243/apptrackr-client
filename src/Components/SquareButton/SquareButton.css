/* Classes */

.squarebutton {
    font-size: 20px;
    padding: 5px 5px;
    border-radius: 8px;
    border: gray;
    width: 30px;
    height: 30px;
    margin-right: 10px;
}

.squarebutton:hover {
    box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24), 0 17px 50px 0 rgba(0,0,0,0.19);
}