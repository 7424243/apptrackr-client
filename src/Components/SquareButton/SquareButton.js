import React from 'react'
import './SquareButton.css'

function SquareButton(props) {
    return (
        <div className='squarebutton_content'>

            <button className='squarebutton' {...props}></button>

        </div>
    )
}

export default SquareButton