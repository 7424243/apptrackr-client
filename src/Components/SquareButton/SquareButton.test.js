import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import SquareButton from './SquareButton'

describe('SquareButton Component', () => {

    const props = {
        path: '/',
        content: 'test'
    }

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<SquareButton {...props}/>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<SquareButton {...props}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

})