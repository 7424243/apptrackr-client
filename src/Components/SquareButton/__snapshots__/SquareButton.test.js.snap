// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareButton Component renders the UI as expected 1`] = `
<div>
  <a
    className="squarebutton_content"
    href="/"
    onClick={[Function]}
  >
    <button
      className="squarebutton"
    >
      test
    </button>
  </a>
</div>
`;
